### ISSUE TITLE
clone_only strategy fail - panics - not implemented

### ISSUE THREAD

**Issue by nphias** [holochain/holochain#3056]
**Describe the bug**
In the App_manifest using the provisioning strategy: clone_only
causes a panic and crash during InstallApp

**Expected behavior**
during InstallApp , should defer the role and allow the agent to clone the DNA at runtime 

**System information:**
 - OS: mac os
 - Holochain and HDK Version 0.2.3-rc.1

**Steps to reproduce**
in the app manifest happ.yaml add a role entry like this:

```
- name: mydna
    provisioning:
      strategy: clone_only
      deferred: true
    dna:
      bundled: "../dnas/mydna/workdir/mydna.dna"
      modifiers:
        network_seed: ~
        properties: ~
        origin_time: ~
        quantum_time: ~
      installed_hash: ~
      clone_limit: 1000
```
then try to run the conductor

error messages:

```
00:25:28 [Tryorama - Local Conductor] info: FATAL PANIC PanicInfo {
    payload: Any { .. },
    message: Some(
        not implemented,
    ),
    location: Location
00:25:28 [Tryorama - Local Conductor] info:  {
        file: "crates/holochain_conductor_api/src/app_interface.rs",
        line: 360,
        col: 17,
    },
    can_unwind: true,
}
```

**Comment by ThetaSinner** [comment]
Thanks for raising this @nphias. My preliminary investigation suggests this is indeed not implemented correctly.

So in this test the code relies on all_cells which ends up looking at role_assignments in here https://github.com/holochain/holochain/blob/develop-0.2/crates/holochain/src/conductor/tests/install_app_bundle.rs#L82

Now following what happens with those role assignments
CloneOnly is recognised as needing the ProvisionOnly strategy https://github.com/holochain/holochain/blob/develop-0.2/crates/holochain_types/src/app/app_bundle.rs#L210
That results in role assignments being updated here https://github.com/holochain/holochain/blob/develop-0.2/crates/holochain_types/src/app/app_bundle.rs#L109. But notice the line above, where dnas_to_register is also set
Next, in the conductor it will ask for cells_to_create https://github.com/holochain/holochain/blob/develop-0.2/crates/holochain/src/conductor/conductor.rs#L1389. That is entirely based on dnas_to_register https://github.com/holochain/holochain/blob/develop-0.2/crates/holochain_types/src/app/app_bundle.rs#L297 and NOT the actual role assignments.
Now the conductor goes ahead and runs genesis for those cells https://github.com/holochain/holochain/blob/develop-0.2/crates/holochain/src/conductor/conductor.rs#L1417
Only after genesis succeeds does the conductor check the actual role assignments to report the conductor state https://github.com/holochain/holochain/blob/develop-0.2/crates/holochain/src/conductor/conductor.rs#L1420

**Comment by ThetaSinner** [comment]
I am nearly done working through the sys_validation workflow which is a priority right now but this is a pretty serious thing to not have working so I'd like to look at this next. @abe-njama for discussion in our meeting later please

**Comment by ThetaSinner** [comment]
I think I've got a fix for this, just waiting to see that it doesn't break anything else but I think that's reasonably unlikely :)

**Comment by ThetaSinner** [comment]
Keeping this open for the backport
